src/win/tty.c

```diff
@@ -2259,25 +2262,95 @@ int uv_tty_reset_mode(void) {
   return 0;
 }
 
-/* Determine whether or not this version of windows supports
- * proper ANSI color codes. Should be supported as of windows
- * 10 version 1511, build number 10.0.10586.
- */
-static void uv__determine_vterm_state(HANDLE handle) {
+static int uv__guess_tty(HANDLE handle)
+{
+  char env_var[5];
   DWORD dwMode = 0;
+  DWORD env_length;
+  int result = UV_TTY_NONE;
+
+  if (handle == INVALID_HANDLE_VALUE || !GetConsoleMode(handle, &dwMode)) {
+    return result;
+  }
+
+  env_length = GetEnvironmentVariableA("ConEmuANSI", env_var, sizeof(env_var));
+  if (env_length == 2 && !strncmp(env_var, "ON", 2)) {
+    HANDLE process_handle = GetCurrentProcess();
+
+    while(1) {
+      NTSTATUS status;
+      PROCESS_BASIC_INFORMATION pbi;
+      ULONG return_length;
+      WCHAR parent_file_name[MAX_PATH];
+      DWORD parent_file_name_length;
+      WCHAR* conemu_file_names[] = { L"\\ConEmu.exe", L"\\ConEmu64.exe" };
+      unsigned int i;
+
+      status = pNtQueryInformationProcess(process_handle,
+                                          ProcessBasicInformation,
+                                          &pbi,
+                                          sizeof(pbi),
+                                          &return_length);
+      CloseHandle(process_handle);
+      if (!NT_SUCCESS(status)) {
+        break;
+      }
 
-  if (!GetConsoleMode(handle, &dwMode)) {
-    uv__vterm_state = UV_UNSUPPORTED;
-    return;
+      process_handle = OpenProcess(PROCESS_QUERY_INFORMATION,
+                                   FALSE,
+                                   pbi.InheritedFromUniqueProcessId);
+      if (!process_handle) {
+        break;
+      }
+
+      parent_file_name_length =
+        GetProcessImageFileNameW(process_handle,
+                                 parent_file_name,
+                                 sizeof(parent_file_name));
```

Comment:
Size should be in characters, not bytes. This could lead to writing past the end of the buffer.